gorma.go

```diff
@@ -120,19 +184,185 @@ func (g *Generator) Generate(api *design.APIDefinition) ([]string, error) {
 		if err != nil {
 			fmt.Println("Error executing Gorma: ", err.Error())
 			g.Cleanup()
-			return g.genfiles, err
+			return err
 		}
 		if err := rbacw.FormatCode(); err != nil {
 			fmt.Println("Error executing Gorma: ", err.Error())
 			g.Cleanup()
-			return nil, err
+			return err
 		}
 		if err == nil {
 			g.genfiles = append(g.genfiles, rbacfilename)
 		}
 	}
 
-	return g.genfiles, err
+	return err
+}
+
+// Generate produces the generated media files
+func (g *Generator) generateResources(api *design.APIDefinition) error {
+	os.MkdirAll(modelDir(), 0755)
+	app := kingpin.New("Model generator", "model generator")
+	codegen.RegisterFlags(app)
+	_, err := app.Parse(os.Args[1:])
+	if err != nil {
+		panic(err)
+	}
+	gopath := filepath.SplitList(os.Getenv("GOPATH"))[0]
+
+	imp, err := filepath.Rel(filepath.Join(gopath, "src"), codegen.OutputDir)
+	if err != nil {
+		return err
+	}
+	imp = path.Join(filepath.ToSlash(imp), "app")
+	imports := []*codegen.ImportSpec{
+		codegen.SimpleImport(imp),
+		codegen.SimpleImport("github.com/jinzhu/copier"),
+	}
+	// get the imports for the app packages
+	api.IterateVersions(func(v *design.APIVersionDefinition) error {
+		if v.IsDefault() {
+			return nil
+		}
+		imports = append(imports, codegen.SimpleImport(imp+"/"+codegen.Goify(codegen.VersionPackage(v.Version), false)))
+		return nil
+	})
+
+	title := fmt.Sprintf("%s: Media Helpers", api.Name)
+
+	err = api.IterateVersions(func(v *design.APIVersionDefinition) error {
+		err = v.IterateResources(func(res *design.ResourceDefinition) error {
+			actionable := false
+			err = res.IterateActions(func(ad *design.ActionDefinition) error {
+				if hasUserType(ad) {
+					actionable = true
+				}
+				return nil
+
+			})
+			if !actionable {
+				return nil
+			}
+
+			if !res.SupportsVersion(v.Version) {
+				return nil
+			}
+			prefix := "resource"
+			if v.Version != "" {
+				prefix = prefix + "v" + codegen.Goify(v.Version, false)
+
+			}
+			name := strings.ToLower(res.Name)
+
+			mediafilename := filepath.Join(modelDir(), name+prefix+"_genmodel.go")
```

Comment:
Can we use `name+"_"+prefix+"_genmodel.go"`?  I use snake_case for my media type ("vnd.compound_name+json"), and the current construct create "compound_nameresource_genmodel.go".